netbox/utilities/forms/bulk_import.py

```diff
@@ -90,8 +90,17 @@ def _clean_csv(self, data):
         """
         Clean CSV-formatted data. The first row will be treated as column headers.
         """
+
+        # Determine the CSV dialect
+        try:
+            # This uses a rough heuristic to detect the CSV dialect. If the data is malformed, we'll fall back to
+            # the default Excel dialect.
+            dialect = csv.Sniffer().sniff(data.strip(), delimiters=',; ')
```

Comment:
It would probably be a good idea to provide the user a mechanism for overriding this as well, like what we do for bulk import formats (CSV/JSON/auto).

Reply:
If you want specific selectors in the form we can get rid of this logic and simply pass the `delimiter` to the reader. Using sniff seemed more manageable in case we decide to include more delimiters or even provide ways for the user to set this in the configuration.

Reply:
We should retain the automatic detection as the default behavior, but also offer the option for the user to override it.

Reply:
I have made some changes to the code, see if that fits what you were articulating. It may need more cleanup though.